### ISSUE TITLE
BabylonJS demo is not compiled with Bridge 16.4

### ISSUE THREAD

**Issue by vladsch** [bridgedotnet/Bridge#3211]
Test project (I was not able to reproduce the issue outside the project)
https://github.com/Retyped/Demos/tree/master/BabylonJsDemo

An exception occurs during compilation 
```
Error: 2017-10-19T19:20:25:047     Error        System.NullReferenceException: Object reference not set to an instance of an object.
   at Bridge.Contract.Rules.Get(IEmitter emitter, IEntity entity)
   at Bridge.Contract.Rules.GetVirtualMemberRules(IEmitter emitter, IEntity entity)
   at Bridge.Contract.Rules.Get(IEmitter emitter, IEntity entity)
   at Bridge.Translator.PreconverterDetecter.VisitMethodDeclaration(MethodDeclaration methodDeclaration)
   at ICSharpCode.NRefactory.CSharp.MethodDeclaration.AcceptVisitor(IAstVisitor visitor)
   at ICSharpCode.NRefactory.CSharp.DepthFirstAstVisitor.VisitChildren(AstNode node)
   at Bridge.Translator.PreconverterDetecter.VisitTypeDeclaration(TypeDeclaration typeDeclaration)
   at ICSharpCode.NRefactory.CSharp.TypeDeclaration.AcceptVisitor(IAstVisitor visitor)
   at ICSharpCode.NRefactory.CSharp.DepthFirstAstVisitor.VisitChildren(AstNode node)
   at ICSharpCode.NRefactory.CSharp.DepthFirstAstVisitor.VisitNamespaceDeclaration(NamespaceDeclaration namespaceDeclaration)
   at ICSharpCode.NRefactory.CSharp.NamespaceDeclaration.AcceptVisitor(IAstVisitor visitor)
   at ICSharpCode.NRefactory.CSharp.DepthFirstAstVisitor.VisitChildren(AstNode node)
   at ICSharpCode.NRefactory.CSharp.DepthFirstAstVisitor.VisitSyntaxTree(SyntaxTree syntaxTree)
   at ICSharpCode.NRefactory.CSharp.SyntaxTree.AcceptVisitor(IAstVisitor visitor)
   at Bridge.Translator.Translator.Preconvert(MemberResolver resolver, IAssemblyInfo config)
   at Bridge.Translator.Translator.Translate()
   at Bridge.Build.BridgeCompilerTask.Execute() (1, 1)
```

**Comment by fabriciomurta** [comment]
Steps to reproduce:
1. Clone the project and update Bridge to 16.4.0 using NuGet package manager.
2. Say 'no' when it asks to replace bridge.json (no new bridge settings are brought from bridge.json during upgrade, so it is safe to be ignored -- and it replaces inherent project settings).
3. Try to build. Build will fail with the error above.

The minimal test case woud be: 
1. Remove all .cs files from project, leaving only:
  - `SceneProviders/AbstractSceneProvider.cs`
  - `SceneProviders/Scene1Provider.cs`
2. change their content to:
  - `AbstractSceneProvider.cs`:
```csharp
using Retyped;

namespace BabylonJsDemo.SceneProviders
{
    public abstract class AbstractSceneProvider
    {
        public abstract int CreateScene(babylon_js.BABYLON.Engine engine);
    }
}
```
  - `Scene1Provider.cs`
```csharp
using static Retyped.babylon_js.BABYLON;

namespace BabylonJsDemo.SceneProviders
{
    public class Scene1Provider : AbstractSceneProvider
    {
        public override int CreateScene(Engine engine)
        {
            int scene = 4;
            return scene;
        }
    }
}
```

If you change the references to `Engine` in both files to `int`, for example, the error will no longer happen on build.

**Comment by fabriciomurta** [comment]
This fix has been just merged into the branch for version 16.4.1.
